### PR TITLE
docs(input): Added Space.Compact recommendations

### LINKED ISSUE #41078: Input.Group size=large or size=small is not honored for the button
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-forked-w13m6t?file=/index.js)

### Steps to reproduce

Create a compact `Input.Group` size set to 'large' or 'small' with one input and one button in it.

### What is expected?

It is expected that the `size` modifier on input group will be honored by buttons inside it.

### What is actually happening?

The sizes 'large' and 'small' are not applied to the button in `Input.Group`. If the size is set to 'large' manually on button, the border radiuses will be incorrect, and input and button will look separate.

| Environment | Info |
| --- | --- |
| antd | 5.3.0 |
| React | 18.2.0 |
| System | Windows 10 build 19045 |
| Browser | Microsoft Edge 110.0.1587.57 |

### PR DESCRIPTION
**Pull request by Yuiai01**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [x] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close https://github.com/ant-design/ant-design/issues/41078

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |  Added `Space.Compact` recommendations          |
| 🇨🇳 Chinese |  添加 `Space.Compact` 的推荐         |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/input/index.en-US.md [hunk 1]
```diff
@@ -96,10 +96,7 @@ Supports all props of `Input`.
 
 #### Input.Group
 
-| Property | Description | Type | Default |
-| --- | --- | --- | --- |
-| compact | Whether use compact style | boolean | false |
-| size | The size of `Input.Group` specifies the size of the included `Input` fields. Available: `large` `default` `small` | string | `default` |
+It is recommended to use [Space.Compact](/components/space#spacecompact), which is more powerful than `Input.Group`. | Property | Description | Type | Default | | --- | --- | --- | --- | | compact | Whether use compact style | boolean | false | | size | The size of `Input.Group` specifies the size of the included `Input` fields. Available: `large` `default` `small` | string | `default` |
```
Comment (kiner-tang) [inline components/input/index.en-US.md:99]: 格式有点奇怪，为什么表格没有换行？

Comment (Yuiai01) [reply components/input/index.en-US.md:99]: 很奇怪，我这边保存好就是这样，导致我没发现，我重新提交一下

**Comment by MadCcc** [comment]
把 demo 里的 Input.Group 换成 Compact，下面的 API 表明一下废弃

File: components/input/index.en-US.md [hunk 2]
```diff
@@ -96,6 +96,8 @@ Supports all props of `Input`.
 
 #### Input.Group
 
+It is recommended to use [Space.Compact](/components/space#spacecompact), which is more powerful than `Input.Group`.
```
Comment (zombieJ) [inline components/input/index.en-US.md:99]: 介绍直接下掉，然后原本的 Input.Group Demo 改成 debug 模式，加一个新的 Demo 直接用的 Compact，然后写上更多请查看 Space.Compact 文档~

Comment (kiner-tang) [reply components/input/index.en-US.md:99]: @Yuiai01 cc

File: components/input/demo/compact-style.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+## zh-CN
+
+使用 `Space.Compact` 替代 `Input.Group`，更多请查看 [Space.Compact](/components/space#spacecompact) 文档。
```
Comment (MadCcc) [inline components/input/demo/compact-style.md:3]: `Input.Group` 相关的描述都可以去掉

Comment (Yuiai01) [reply components/input/demo/compact-style.md]: 嗯嗯，这部分删除掉了～

**Review by afc163 — APPROVED** [review]

**Pull request merged**